### PR TITLE
Configuring Session Certificates for Dynamic Networks

### PR DESCRIPTION
**Pull request by nadinequinr3**

### CODE REVIEW

File: content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/session-certificates.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
+---
+date: '2022-11-15'
+title: "Configuring Optional Session Certificates"
+menu:
+  corda-5-alpha:
+    identifier: corda-5-alpha-mgm-onboarding
```
Comment (charlieR3) [inline content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/session-certificates.md:6]: I'm not too familiar with these headers but these seems like it's referencing the wrong page

File: content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/session-certificates.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+3. Send the following request to disable revocation checks for the specified gateway worker:
+   ```
+   curl -k -u admin:admin -X PUT -d '{"section":"corda.p2p.linkManager", "version":"'$CONFIG_VERSION'", "config": "{ "revocationCheck": { "mode": "OFF" } }", "schemaVersion": {"major": 1, "minor": 0}}' $API_URL"/config"
+   ```
+
+## Build Registration Context for MGM Registration
+
+If using session certificates, you must also specify the following in the [MGM REGISTRATION_CONTEXT](mgm-onboarding.html#build-registration-context):
+
+* Add an extra JSON field `corda.group.truststore.session.0` with the truststore of the CA to the registration context (similar to `corda.group.truststore.tls.0`).
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/session-certificates.md:69]: Should these two bullets points be numbered bullets?

**Review by paulmoloneyr3 — CHANGES_REQUESTED** [review]
Just a note on the bullet points

File: content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/session-certificates.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+```
+Similarly to the TLS certificate, the CSR can be processed to issue a certificate using either a real CA of your choice or the fake CA dev tool. To use the fake CA tool:
+```shell
+cd $RUNTIME_OS
+java -jar ./applications/tools/p2p-test/fake-ca/build/bin/corda-fake-ca-5.0.0.0-SNAPSHOT.jar -m /tmp/ca csr $WORK_DIR/request.csr
+cd $WORK_DIR
+```
+This command outputs the location of the signed certificate. For example:
+```shell
+Wrote certificate to /tmp/ca/request1/certificate.pem
```
Comment (charlieR3) [inline content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/session-certificates.md:35]: I think we should remove mention of the fake CA from the public docs.

```
Similarly to the TLS certificate, the CSR can be processed to issue a certificate using a CA chosen by the MGM operator. The CA trustroot for session certificates should be configured during the MGM onboarding.
```

File: content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/session-certificates.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+```
+At this point, you should already have a certificate based on the CSR exported from Corda issued either by a real CA or by the fake CA tool. You must upload the certificate chain to the Corda cluster. To upload the certificate chain, run:
```
Comment (charlieR3) [inline content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/session-certificates.md:37]:
```suggestion
Once you have a certificate based on the CSR exported from Corda issued by the CA, you must upload the certificate chain to the Corda cluster. To upload the certificate chain, run:
```

File: content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/session-certificates.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+You can optionally omit the root certificate.
+
+{{< note >}}
+If you upload a certificate chain consisting of more than one certificates, ensure that `-----END CERTIFICATE-----` and `-----BEGIN CERTIFICATE-----` from the next certificate are separated by a new line with no empty spaces in between.
+{{< /note >}}
+
+### Disable Revocation Checks
+
+If the CA has not been configured with revocation (for example, via CRL or OCSP), you can disable revocation checks. By default, revocation checks are enabled.
+The fake CA dev tool does not support revocation and so, if you are using the fake CA, you must disable revocation checks.
```
Comment (charlieR3) [inline content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/session-certificates.md:50]:
```suggestion

```

File: content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/session-certificates.md [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+2. Save the displayed version number from the response as a variable:
+   ```shell
+   export CONFIG_VERSION=<configuration-version>
+   ```
+3. Send the following request to disable revocation checks for the specified gateway worker:
+   ```
+   curl -k -u admin:admin -X PUT -d '{"section":"corda.p2p.linkManager", "version":"'$CONFIG_VERSION'", "config": "{ "revocationCheck": { "mode": "OFF" } }", "schemaVersion": {"major": 1, "minor": 0}}' $API_URL"/config"
+   ```
+
+## Build Registration Context for MGM Registration
```
Comment (charlieR3) [inline content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/session-certificates.md:65]: I would move this section to the top after `## Set Variables`.
This section is in regards to the MGM onboarding, whereas the rest of the doc is in regards to member onboarding. 
It think we should link to this from the "build registration context" of the MGM onboarding doc also.

File: content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/session-certificates.md [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+
+If using session certificates, you must also specify the following in the [MGM REGISTRATION_CONTEXT](mgm-onboarding.html#build-registration-context):
+
+* Add an extra JSON field `corda.group.truststore.session.0` with the truststore of the CA to the registration context (similar to `corda.group.truststore.tls.0`).
+* Set the JSON field `corda.group.pki.session` to `"Standard"` instead of `"NoPKI"`.
+
+## Configure Virtual Node as Network Participant
+
+If using session certificates, you must also add the `sessionCertificateChainAlias` JSON field to the network setup RPC request. For example:
```
Comment (charlieR3) [inline content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/session-certificates.md:74]:
```suggestion
If using session certificates, you must also add the `sessionCertificateChainAlias` and `useClusterLevelSessionCertificateAndKey` JSON fields to the network setup RPC request. For example:
```

File: content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/session-certificates.md (hunk 1 above)
Comment (nadinequinr3) [reply content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/session-certificates.md:6]: > I'm not too familiar with these headers but these seems like it's referencing the wrong page

good spot, thanks!

**Review by colin-hillis — APPROVED** [review]
Bar Charlie's technical comments still to be added, this LGTM

**Review by williamvigorr3 — COMMENTED** [review]
Looks good.

**Review by paulmoloneyr3 — APPROVED** [review]

**Review by colin-hillis — APPROVED** [review]
LGTM

**Pull request merged**